### ISSUE TITLE
Spotify Password Resets

### ISSUE THREAD

**Issue by n8klayko** [justin025/onthespot#86]
I installed and started using this yesterday with my Spotify Premium account and it caused my account password to be reset by Spotify. I used the default settings and after about 15-20 songs it stopped downloading. I wasn't sure why nothing would download anymore, but fast forward a few hours and it turns out Spotify reset my password for suspicious activity. I used the default settings only changing a few things like download location and naming scheme. I am wondering if this issue is something others have experienced or if there are more recommended settings so that things work without flagging suspicious activity to Spotify?

For Reference, I am using build 1.0.4 and with 1 queue worker, 1 download worker, and a download delay of 3. I did not initially try changing any settings from how it was at default, but I noticed the "Mirror Spotify playback option". Does this option help to elminate the issue I experienced or should I be doing something different?

**Comment by justin025** [comment]
The password resets are normal now and then just reset and login again. For failed downloads setting the download delay to 35 seconds is necessary for only spotify otherwise downloads will fail. Mirror spotify playback will download the currently playing song on your spotify app. Hope this helps

**Comment by legaldeejay** [comment]
My experience with Spotify downloaders such as this program is that when you download a song, the song is actually playing silently in Spotify at a much faster rate. If you are old enough, you may remember the good old days of dubbing with dual cassette decks where you can record or dub content from one cassette onto a blank one at 2X the speed.   This is similar to what is happening here.  When Spotify detects this, it flags and suspends the account, although not necessarily right away.  So I recommend that you use a free Spotify account instead when doing this.  If the free account gets suspended, you can always open a new one.   This happens to me all the time and I just open a new account when it happens.   

Maybe Justin can add a feature to adjust the speed of the download which could help prevent the issue, except it will obviously take much longer.

**Comment by n8klayko** [comment]
Thank you both for your responses. I sort of figured this was normal behavior. I was using an app that used a Spotify premium account with developer access to download essentially without restrictions, but it stopped working a few months ago and it appears Spotify has broken that method now. I was hoping to use an app that downloads with the premium account to get the lossless audio. Using a free account appears to be limited to 160kbps with this app. I used to rip things that way back in the day, but the main reason I want to download is for tracks and albums that are rare or that I can't purchase their music because its not available DRM-free. Everybody streams everything and nobody owns anything anymore which is super annoying... anyways rant over.

I'd actually love the ability to slow down the download speed to prevent password resets just so that I could queue up a bunch of stuff to download overnight.

**Comment by legaldeejay** [comment]
I do not believe free accounts are limited to 160k.  In the Settings screen, it says Max Bitrate 160k in the Account section, but in the Settings, you can set the File Bitrate to 320k.   Using an acoustic spectrum analyzer, it appears the file is a true 320k.  Maybe Justin can explain this further.

**Comment by justin025** [comment]
Just to clear up a few misconceptions, and for future documentation, onthespot does not play songs at a faster than normal rate and then record them, or play songs at all for that matter. Spotify returns the entire ogg file all at once and, once the file is saved to your system, the application will wait a set amount of time before attempting the next download. The download delay mimics a user listening to a track before streaming the next, if you want spotify to think you're listening to the whole 3 minutes (assuming your track is 3 minutes long) set the download delay to 180 seconds. Spotify has had rolling rate limit windows for a number of years now and you can even hit rate limits in their native apps just by skipping songs really fast, you will receive a popup telling you to slow down, after a maximum of 30 minutes your rate limit will be lifted and you can continue streaming / downloading. Many users have had success with a 35 second download delay reporting no failed downloads, I have left the app running for hours with a 35 second delay without any errors, you could try that for overnight downloads.

The reason spotify will reset your password for suspicious activity is because the way onthespot retrieves your credentials is abnormal (it normally used for external devices), and it is even less common to then begin streaming hundreds of songs on said external device. It's just a safety measure, in the same way if you login to spotify on airport wifi you're bound to get a similar reset password email once you get home. But once you have reset your password and logged in to said device again, spotify will accept that device's traffic as normal. My premium account hasn't received a reset password email for months but when i started developing the app i received maybe 2 or 3. You can ask around on the community discord I can assure you that none of the users have received a ban thus far, and if someone had I would include a bright red disclaimer.

>I was hoping to use an app that downloads with the premium account to get the lossless audio.

Spotify Premium does not provide lossless audio, it returns a 320kbps ogg. For lossless tracks consider deezer or tidal.

>I do not believe free accounts are limited to 160k. In the Settings screen, it says Max Bitrate 160k in the Account section, but in the Settings, you can set the File Bitrate to 320k. Using an acoustic spectrum analyzer, it appears the file is a true 320k. Maybe Justin can explain this further.
ffmpeg converts your file from ogg to mp3 (or whatever output you specify) but in some instances the user may want to specify their output bitrate instead of letting ffmpeg guess what is best. The file bitrate setting will upscale or downscale the downloaded file to whatever is set, this is necessary to downscale say a flac to an mp3 but can be confusing when upscaling a 160k mp3 to 320k. I can assure you that although your spectrum analyzer says 320k it does not have any more data than the original 160k file had.

I hope this helps out

—

**Comment by Pluvillion** [comment]
The perfect answer. I'm subbed to Premium but I also want to use some songs (school-related) without having to dig into sites which makes it a hassle to do (Lucida is slow, free-mp3-download is gone, and I'd be damned to use a converted YT video). I've been scouring the internet just to know if using OnTheSpot is safe. I finally got the answer from the man himself. Thank you, Justin.
